zirgen/Dialect/BigInt/Bytecode/test/test.cpp

```diff
@@ -0,0 +1,205 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/DialectRegistry.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/Pass/PassManager.h"
+#include "zirgen/Dialect/BigInt/Bytecode/decode.h"
+#include "zirgen/Dialect/BigInt/Bytecode/encode.h"
+#include "zirgen/Dialect/BigInt/Bytecode/file.h"
+#include "zirgen/Dialect/BigInt/IR/BigInt.h"
+#include "zirgen/Dialect/BigInt/IR/Eval.h"
+#include "zirgen/Dialect/BigInt/Transforms/Passes.h"
+#include "zirgen/circuit/bigint/op_tests.h"
+#include "zirgen/circuit/bigint/rsa.h"
+#include "llvm/ADT/APInt.h"
+
+#include <gtest/gtest.h>
+
+using namespace zirgen;
+
+using ZType = std::array<uint32_t, 4>;
+
+struct BibcTest : public testing::Test {
+  std::unique_ptr<mlir::MLIRContext> context;
+  mlir::MLIRContext* ctx;
+  mlir::ModuleOp module;
+
+  BibcTest() {
+    mlir::DialectRegistry registry;
+    registry.insert<BigInt::BigIntDialect>();
+    registry.insert<mlir::func::FuncDialect>();
+
+    context = std::make_unique<mlir::MLIRContext>(registry);
+    context->loadAllAvailableDialects();
+    ctx = context.get();
+
+    auto loc = mlir::UnknownLoc::get(ctx);
+    module = mlir::ModuleOp::create(loc);
+  }
+
+  mlir::func::FuncOp makeFunc(std::string name, mlir::OpBuilder& builder) {
+    auto loc = mlir::UnknownLoc::get(ctx);
+    builder.setInsertionPointToEnd(&module.getBodyRegion().front());
+    auto funcType = mlir::FunctionType::get(ctx, {}, {});
+    auto out = builder.create<mlir::func::FuncOp>(loc, name, funcType);
+    builder.setInsertionPointToEnd(out.addEntryBlock());
+    builder.create<mlir::func::ReturnOp>(loc);
+    builder.setInsertionPointToStart(builder.getInsertionBlock());
+    return out;
+  }
+
+  mlir::func::FuncOp recycle(mlir::func::FuncOp inFunc) {
+    // Encode this function into BIBC structure
+    auto prog = BigInt::Bytecode::encode(inFunc);
+    // Write it out into a buffer
+    size_t bytes = BigInt::Bytecode::tell(*prog);
+    auto buf = std::make_unique<uint8_t[]>(bytes);
+    BigInt::Bytecode::write(*prog, buf.get(), bytes);
+    // Drop the old bytecode structure and create a fresh one
+    prog.reset(new BigInt::Bytecode::Program);
+    // Read the contents of the buffer back in
+    BigInt::Bytecode::read(*prog, buf.get(), bytes);
+    // Decode the bytecode back into MLIR operations
+    return BigInt::Bytecode::decode(module, *prog);
+  }
+
+  void lower() {
+    // Lower the inverse and reduce ops to simpler, executable ops
```

Comment:
Did you forget to add `createLowerInvPass` in this function? Or is there a reason it's not here?

Reply:
The RSA test cases work fine without it. This test apparatus would certainly be more generally useful if it were included, though, so I'll add that call.